### PR TITLE
Curt branch

### PR DESCRIPTION
**Pull request by ckolovson**
Added update_catalog function and unit test.

### CODE REVIEW

File: pyvcloud/vcd/org.py [hunk 1]
```diff
@@ -107,6 +107,28 @@ def get_catalog(self, name):
                 return self.client.get_resource(link.href)
         raise Exception('Catalog not found.')
 
+    def update_catalog(self, old_catalog_name, new_catalog_name, description):
```
Comment (anusuyar) [inline pyvcloud/vcd/org.py]: Can you add docstring for this method.

Comment (ckolovson) [reply pyvcloud/vcd/org.py]: I added a comment before the method. Is that what you meant by "docstring"?

Comment (anusuyar) [reply pyvcloud/vcd/org.py]: No. it has to be after the method name. You can follow the docstring format present in pyvcloud/vdc.py

Comment (ckolovson) [reply pyvcloud/vcd/org.py]: OK, done.

File: pyvcloud/vcd/org.py [hunk 2]
```diff
@@ -107,6 +107,35 @@ def get_catalog(self, name):
...
+        """
+        Update the name and/or description of a catalog.
+        :param old_catalog_name: (str): The current name of the catalog.
+        :param new_catalog_name: (str): The new name of the catalog.
+        :param description: (str): The new description of the catalog.
+        :return:  A :class:`lxml.objectify.StringElement` object describing the updated catalog.
+        """
+        org = self.client.get_resource(self.href)
```
Comment (rocknes) [inline pyvcloud/vcd/org.py]: if self.resource is None:
             self.resource = self.client.get_resource(self.href)

Will save an extra ReST call to vCD if Org object is already initialized.

File: pyvcloud/vcd/org.py [hunk 3]
```diff
@@ -107,6 +107,35 @@ def get_catalog(self, name):
...
+        links = get_links(org,
+                          rel=RelationType.DOWN,
+                          media_type=EntityType.CATALOG.value)
+        for link in links:
+            if old_catalog_name == link.name:
+                catalog = self.client.get_resource(link.href)
+                href = catalog.get('href')
+                admin_href = href.replace('/api/catalog/', '/api/admin/catalog/')
+                catalog = self.client.get_resource(admin_href)
```
Comment (rocknes) [inline pyvcloud/vcd/org.py]: Probably it's a good idea to highlight that the new catalog object we get as a result of the admin call is not the same as the old catalog object, but rather a admin view of the catalog. I think we should use a new variable say adminViewOfCatalog.

File: tests/vcd_catalog_update.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    def test_create_catalog(self):
+        logged_in_org = self.client.get_org()
+        org = Org(self.client, resource=logged_in_org)
+        catalog = org.create_catalog(self.config['vcd']['catalog'], 'test catalog')
+        assert self.config['vcd']['catalog'] == catalog.get('name')
+
+    def test_update_catalog(self):
+        logged_in_org = self.client.get_org()
+        org = Org(self.client, resource=logged_in_org)
+        catalog = org.update_catalog(self.config['vcd']['catalog'], self.config['vcd']['new_name'], self.config['vcd']['new_desc'])
```
Comment (rocknes) [inline tests/vcd_catalog_update.py]: Line too long add a #  NOQA at the end of the line.

Comment (anusuyar) [reply tests/vcd_catalog_update.py]: You can probably make the line into 2. I think the line length limit is 80 chars.

**Comment by anusuyar** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vmware/pyvcloud/122)*

**Comment by ckolovson** [comment]
In update_catalog, there is this code sequence:
---
if self.resource is None:
    self.resource = self.client.get_resource(self.href)
org = self.client.get_resource(self.href)
---
Should the third line be replaced by:
org = self.resource
?

**Pull request merged**